### PR TITLE
Wrench but better

### PR DESCRIPTION
**Pull request by NCBPFluffyBear**
## Short Description
Added wrench that instantly breaks machines. On a server with `need-wrench-to-break-machines` set to false, the wrench has a 0% failure chance and breaking machines without the wrench will drop the machine. On a server with `need-wrench-to-break-machines` set to true, the wrench has a `wrench-failure-chance` (default of 0.3) to fail when breaking machines, and if a machine is broken without a wrench, it will drop a machine block. Wrench takes default durability damage when used, can be used by left and right clicking machines, and only works in main hand.

*Machines include generators but do not include capacitors or cargo nodes
https://streamable.com/3wd61s (This is a new video)

**This changes the old wrench by simply calling a block break event for cargo, which is simpler and more efficient.

## Additions/Changes/Removals
- Added wrench class
- Added wrench item lore
- Added wrench registering
- Added `need-wrench-to-break-machines` config option (true for new servers, false for existing servers)
- Added `wrench-failure-chance` config option (0.0 for new servers, 0.0 for existing servers)
- Added block break event that turns machines and generators into machine blocks when broken without wrench

## Related Issues
#52 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/items/Wrench.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ *
+ * @author FluffyBear
+ */
+public class Wrench extends SimpleSlimefunItem<ItemUseHandler> implements Listener, DamageableItem {
+
+    public Wrench() {
+        super(Items.LITEXPANSION, Items.WRENCH, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            Items.REFINED_IRON, Items.REFINED_IRON, Items.REFINED_IRON,
+            new ItemStack(Material.SHEARS), Items.ADVANCED_CIRCUIT, new ItemStack(Material.SHEARS),
+            null, Items.CARBON_PLATE, null
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/Wrench.java]: Default wrench should be just copper

File: src/main/java/dev/j3fftw/litexpansion/items/Wrench.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        Bukkit.getPluginManager().registerEvents(this, LiteXpansion.getInstance());
+    }
+
+    @Nonnull
+    public ItemUseHandler getItemHandler() {
+        return e -> e.setUseBlock(Event.Result.DENY);
+    }
+
+    @EventHandler
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/Wrench.java]: We still have an event class

File: src/main/java/dev/j3fftw/litexpansion/items/Wrench.java (hunk 1 above)
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/items/Wrench.java]: whoopsies forgot to change that

**Comment by J3fftw1** [comment]
Add research

**Comment by J3fftw1** [comment]
conflicts

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 3]
```diff
@@ -68,6 +88,111 @@ public void onHungerDamage(EntityDamageEvent e) {
...
+                    double random = Math.random();
+                    wrenchBlock(p, block, random <= Constants.WRENCH_FAIL_CHANCE, true);
+
+                } else {
+                    wrenchBlock(p, block, false, true);
+                }
+
+                wrench.damageItem(p, playerWrench);
+
+            } else if (sfBlock != null && (sfBlock.getID().startsWith("CARGO_NODE")
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: move && to new line

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 4]
```diff
@@ -68,6 +88,111 @@ public void onHungerDamage(EntityDamageEvent e) {
...
+                wrenchBlock(p, block, false, true);
+                wrench.damageItem(p, playerWrench);
+
+            } else {
+                Utils.send(p, "&cYou can not use the wrench on this block!");
+            }
+        }
+    }
+
+    public static void wrenchBlock(Player p, Block block, boolean fail, boolean byInteract) {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: little inconsistent here uses Player p and Block b

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Pull request merged**